wasm/module.go

```diff
@@ -28,20 +28,57 @@ func (r *reader) Read(p []byte) (n int, err error) {
 // Module is a WebAssembly binary representation.
 // See https://www.w3.org/TR/wasm-core-1/#modules%E2%91%A8
 type Module struct {
-	TypeSection     []*FunctionType
-	ImportSection   []*ImportSegment
+	// TypeSection contains the unique FunctionType of functions imported or defined in this module.
+	// See https://www.w3.org/TR/wasm-core-1/#type-section%E2%91%A0
+	TypeSection []*FunctionType
+	// ImportSection contains any types, tables, memories or globals imported into this module.
+	// See https://www.w3.org/TR/wasm-core-1/#import-section%E2%91%A0
+	ImportSection []*ImportSegment
+	// FunctionSection contains the index in TypeSection of each function defined in this module.
+	//
+	// Function indexes are offset by any imported functions because the function index space begins with imports,
+	// followed by ones defined in this module. Moreover, the FunctionSection is index correlated with the CodeSection.
+	//
+	// For example, if there are two imported functions and three defined in this module, we expect the CodeSection to
+	// have a length of three and the function at index 3 to be defined in this module. Its type would be at
+	// TypeSection[FunctionSection[0]], while its locals and body are at CodeSection[0].
+	//
+	// See https://www.w3.org/TR/wasm-core-1/#function-section%E2%91%A0
 	FunctionSection []uint32
-	TableSection    []*TableType
-	MemorySection   []*MemoryType
-	GlobalSection   []*GlobalSegment
-	ExportSection   map[string]*ExportSegment
-	// StartSection is a function index value when set.
+	// TableSection contains each table defined in this module.
+	//
+	// Table indexes are offset by any imported tables because the table index space begins with imports, followed by
```

Comment:
maybe worth noting in MVP, multiple tables are not supported!

Reply:
ah right. and that MVP is spec 1.0 😎